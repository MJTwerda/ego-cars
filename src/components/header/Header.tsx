import { Box } from "@mui/material";
import Image from "next/image";
import egoCarsLogo from '../../../public/ego-cars-logo.png';
import CommonMenu from "../menu/Menu";
import { MenuItemI } from "../menu/menu.interfaces";
import CommonTabs from "../tabs/Tabs";
import { TabValue } from '../tabs/tabs.interfaces';

const Header = () => {

  // TODO: Llevar esta data a un archivo aparte
  const headerTabs =[
    {
    value: TabValue.MODELS,
    label: 'Modelos'
    }, {
    value: TabValue.MODELS_TAB,
    label: 'Ficha de Modelo'
  }];

  const menuItems: MenuItemI[] = [
    {
      label: 'Modelos',
      path: 'models',
    },
    {
      label: 'Servicios y Accesorios',
      path: 'services-and-accessories',
    },
    {
      label: 'Financiación',
      path: 'financing',
    },
    {
      label: 'Reviews y Comunidad',
      path: 'reviews-and-community',
      bottomDivider: true,
    },
    {
      label: 'Toyota Mobility Service',
      path: 'toyota-1',
    },
    {
      label: 'Toyota Gazoo Racing',
      path: 'toyota-2',
    },
    {
      label: 'Toyota Híbridos',
      path: 'toyota-hybrid',
      bottomDivider: true,
    },
    {
      label: 'Concesionarios',
      path: 'concessions',
    },
    {
      label: 'Test Drive',
      path: 'test-drive',
    },
    {
      label: 'Contacto',
      path: 'contact',
      bottomDivider: true,
    },
    {
      label: 'Actividades',
      path: 'activities',
    },
    {
      label: 'Servicios al Cliente',
      path: 'client-services',
    },
    {
      label: 'Ventas Especiales',
      path: 'special-sales',
    },
    {
      label: 'Innovación',
      path: 'innovation',
    },
    {
      label: 'Prensa',
      path: 'press',
    },
    {
      label: 'Acerca de...',
      path: 'about',
    }
  ];

  return (
    <div className="flex justify-between m-4">
      <div className="flex w-2/4">
        <Image
          alt='Ego Cars logo'
          src={'/ego-logo.svg'}
          width={38}
          height={40}
        />
        <div className="hidden sm:flex">
          <CommonTabs tabList={headerTabs} />
        </div>
      </div>

      <CommonMenu menuItems={menuItems} />
    </div>
  )
};

export default Header;