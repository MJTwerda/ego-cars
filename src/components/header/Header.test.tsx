import { act, render, RenderResult, waitFor } from "@testing-library/react";
import { useRouter } from "next/navigation";
import Header from "./Header";

jest.mock("next/navigation", () => ({
  useRouter: jest.fn(),
  usePathname: jest.fn(),
}));

const mockPush = jest.fn();
(useRouter as jest.Mock).mockReturnValue({ push: mockPush });

describe('Header', () => {
  beforeEach(() => {
    jest.clearAllMocks();
  });

  it('should render', async () => {
    let component: RenderResult;
    await act(async () => {
      component = render(<Header />);
    });

    await waitFor(() => {
      expect(component).toMatchSnapshot();
    });
  });
});