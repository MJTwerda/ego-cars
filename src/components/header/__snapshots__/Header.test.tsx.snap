// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header should render 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="flex justify-between m-4 border-b-2 border-b-opacity-40"
      >
        <div
          class="flex sm:w-2/4"
        >
          <img
            alt="Ego Cars logo"
            class="sm:mr-20"
            data-nimg="1"
            decoding="async"
            height="40"
            loading="lazy"
            src="/ego-logo.svg"
            style="color: transparent;"
            width="38"
          />
          <div
            class="hidden sm:flex"
          >
            <div
              class="MuiTabs-root css-jhczcq-MuiTabs-root"
            >
              <div
                class="MuiTabs-scroller MuiTabs-fixed css-s2t35c-MuiTabs-scroller"
                style="overflow: hidden; margin-bottom: 0px;"
              >
                <div
                  class="MuiTabs-flexContainer css-1wxkzlj-MuiTabs-flexContainer"
                  role="tablist"
                >
                  <a
                    aria-selected="true"
                    class="MuiButtonBase-root MuiTab-root MuiTab-textColorSecondary Mui-selected css-16iolqj-MuiButtonBase-root-MuiTab-root"
                    data-testid="navigation-tab-/"
                    id="navigation-tab-/"
                    role="tab"
                    tabindex="0"
                  >
                    Modelos
                  </a>
                  <a
                    aria-selected="false"
                    class="MuiButtonBase-root MuiTab-root MuiTab-textColorSecondary css-16iolqj-MuiButtonBase-root-MuiTab-root"
                    data-testid="navigation-tab-model-sheet"
                    id="navigation-tab-model-sheet"
                    role="tab"
                    tabindex="-1"
                  >
                    Ficha de Modelo
                  </a>
                </div>
                <span
                  class="MuiTabs-indicator css-ehhs8s-MuiTabs-indicator"
                  style="left: 0px; width: 0px;"
                />
              </div>
            </div>
          </div>
        </div>
        <div>
          <div
            class="flex justify-center"
          >
            <p
              class="MuiTypography-root MuiTypography-body1 flex self-center w-3/4 css-rizt0-MuiTypography-root"
            >
              Menu
            </p>
            <button
              class="MuiButtonBase-root MuiIconButton-root MuiIconButton-sizeMedium css-53g0n7-MuiButtonBase-root-MuiIconButton-root"
              data-testid="common_menu_icon_btn"
              tabindex="0"
              type="button"
            >
              <svg
                aria-hidden="true"
                class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-1umw9bq-MuiSvgIcon-root"
                data-testid="MenuIcon"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="M3 18h18v-2H3zm0-5h18v-2H3zm0-7v2h18V6z"
                />
              </svg>
            </button>
          </div>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="flex justify-between m-4 border-b-2 border-b-opacity-40"
    >
      <div
        class="flex sm:w-2/4"
      >
        <img
          alt="Ego Cars logo"
          class="sm:mr-20"
          data-nimg="1"
          decoding="async"
          height="40"
          loading="lazy"
          src="/ego-logo.svg"
          style="color: transparent;"
          width="38"
        />
        <div
          class="hidden sm:flex"
        >
          <div
            class="MuiTabs-root css-jhczcq-MuiTabs-root"
          >
            <div
              class="MuiTabs-scroller MuiTabs-fixed css-s2t35c-MuiTabs-scroller"
              style="overflow: hidden; margin-bottom: 0px;"
            >
              <div
                class="MuiTabs-flexContainer css-1wxkzlj-MuiTabs-flexContainer"
                role="tablist"
              >
                <a
                  aria-selected="true"
                  class="MuiButtonBase-root MuiTab-root MuiTab-textColorSecondary Mui-selected css-16iolqj-MuiButtonBase-root-MuiTab-root"
                  data-testid="navigation-tab-/"
                  id="navigation-tab-/"
                  role="tab"
                  tabindex="0"
                >
                  Modelos
                </a>
                <a
                  aria-selected="false"
                  class="MuiButtonBase-root MuiTab-root MuiTab-textColorSecondary css-16iolqj-MuiButtonBase-root-MuiTab-root"
                  data-testid="navigation-tab-model-sheet"
                  id="navigation-tab-model-sheet"
                  role="tab"
                  tabindex="-1"
                >
                  Ficha de Modelo
                </a>
              </div>
              <span
                class="MuiTabs-indicator css-ehhs8s-MuiTabs-indicator"
                style="left: 0px; width: 0px;"
              />
            </div>
          </div>
        </div>
      </div>
      <div>
        <div
          class="flex justify-center"
        >
          <p
            class="MuiTypography-root MuiTypography-body1 flex self-center w-3/4 css-rizt0-MuiTypography-root"
          >
            Menu
          </p>
          <button
            class="MuiButtonBase-root MuiIconButton-root MuiIconButton-sizeMedium css-53g0n7-MuiButtonBase-root-MuiIconButton-root"
            data-testid="common_menu_icon_btn"
            tabindex="0"
            type="button"
          >
            <svg
              aria-hidden="true"
              class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-1umw9bq-MuiSvgIcon-root"
              data-testid="MenuIcon"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M3 18h18v-2H3zm0-5h18v-2H3zm0-7v2h18V6z"
              />
            </svg>
          </button>
        </div>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
